pkg/cloud/azure/storageauthorizer.go

```diff
@@ -127,3 +145,77 @@ func (ah *AuthorizerHolder) GetBlobClient(serviceURL string) (*azblob.Client, er
 func (ah *AuthorizerHolder) UnmarshalJSON(b []byte) error {
 	return json.Unmarshal(b, ah.Authorizer)
 }
+
+type StorageConnectionStringCredential struct {
+	StorageConnectionString string             `json:"storageConnectionString"`
+	HTTPConfig              storage.HTTPConfig `json:"httpConfig"`
+}
+
+func (s *StorageConnectionStringCredential) MarshalJSON() ([]byte, error) {
+	fmap := make(map[string]any, 3)
+	fmap[cloud.AuthorizerTypeProperty] = StorageConnectionStringAuthorizerType
+	fmap["storageConnectionString"] = s.StorageConnectionString
+	fmap["httpConfig"] = s.HTTPConfig
+	return json.Marshal(fmap)
+}
+
+func (s *StorageConnectionStringCredential) UnmarshalJSON(b []byte) error {
```

Comment:
I am wondering if this unmarshalJSON would be necessary if the httpConfig was a pointer which and the default was used when it is nil in `GetBlobClient()`